solarforecastarbiter/pvmodel.py

```diff
@@ -0,0 +1,340 @@
+"""
+Calculate AC power and modeling intermediates from system metadata,
+times, and weather data.
+
+Steps are:
+
+1. Calculate solar position using solar_position
+2. If not already known, calculate 3 irradiance components from measured
+   GHI using irradiance_components or modeled clear sky using clearsky.
+3. calculate_poa_effective
+4. calculate_power
+
+Steps 3 and 4 are bundled in :py:func:`irradiance_to_power`
+"""
+
+from functools import partial
+
+import pvlib
+
+from solarforecastarbiter import datamodel
+
+
+def calculate_solar_position(latitude, longitude, elevation, times):
```

Comment:
`validator.get_solarposition` and `pvmodel.calculate_solar_position` do the same thing.